test/Faker/Provider/kk_KZ/PersonTest.php

```diff
@@ -19,13 +20,28 @@ public function setUp()
 
     public function testIndividualIdentificationNumberIsValid()
     {
-        $birthDate                      = new \DateTime('now');
+        $birthDate                      = DateTime::dateTimeBetween('-30 years', '-10 years');
         $individualIdentificationNumber = $this->faker->individualIdentificationNumber($birthDate);
-        $birthDateAsString              = $birthDate->format('ymd');
+        $sum                            = 0;
 
-        $this->assertRegExp(
-            "/^(" . $birthDateAsString . ")([1-6]{1})(\\d{5})$/",
-            $individualIdentificationNumber
-        );
+        for ($i = 0; $i <= 10; $i++) {
```

Comment:
don't duplicate the checksum code here. Instead, refactor it into a utility function, and use it both in the generator and in the test.